pkg/api/defaults.go

```diff
@@ -480,12 +480,14 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
 			}
 		}
 
-		// Configure kubelet
-		cs.setKubeletConfig(isUpgrade)
-
 		// Configure addons
+		// This needs to be configured first as set kubelet config depends
+		// on certains default addons to be populated (eg. ip-masq-agent)
 		cs.setAddonsConfig(isUpgrade)
 
+		// Configure kubelet
+		cs.setKubeletConfig(isUpgrade)
```

Comment:
Based on https://github.com/Azure/aks-engine/pull/2393/files#r355038175, was there a reason for changing kubelet config defaults to before addons defaults? ie. is there a dependency introduced in #2393 between addons and kubelet config defaults?

Reply:
I think this is one example: https://github.com/Azure/aks-engine/blob/master/pkg/api/addons.go#L576

Reply:
`--cluster-domain` is always set to `cluster.local`. So not sure if we need this change.

Reply:
`ip-masq-agent` addon is enabled by default. It's only explicitly disabled by setting kubernetes config/ hosted master. So instead of checking if the addon is enabled, I've changed the check to see if it's explicitly disabled and then perform the logic. 

@CecileRobertMichon lmk what you think?